### DISCUSSION TITLE
Should we start using live samples in the JS documentation?

### DISCUSSION THREAD

**Discussion by wbamberg** [mdn/mdn-community#617] [Content]
Although we use live samples in most of the core reference bits of MDN (API, CSS, HTML) we don't use them in the JS docs. This PR https://github.com/mdn/content/pull/14128 added one, and I wondered if people thought we should have a project of "live-ifying" examples in the JS reference.

**Comment by alattalatta** [comment]
Though I'm the author of #14128, current way of `{{EmbedLiveSamle}}` will be quick to become unmaintainable because every example would have its own implementation for its output display. An inline equivalent of `{{EmbedInteractiveExample}}` seems necessary, with prebuilt run button and an output area. So, as you mentioned, https://github.com/openwebdocs/project/issues/59?

Eheh, assuming this topic is strictly for JS, not APIs. I can't imagine how one can make it work for an API.

**Reply by wbamberg** [reply]
Yeah, we already use live samples very widely in Web/API.

It's indeed a good point about output. In Web/API (and various other places) this is kind of a problem already, where people choose all sorts of ways to log output: some write their own in-page log, some use `console.log()`, some even use `alert()`.

But it's perhaps more of a problem for JS, where the only output is logging output.

I wonder if it's a feature that could be somehow built into the existing `{{EmbedLiveSample}}` system... I think noone wants to create yet another way to write examples, but adding a new feature to an existing one might be more acceptable.

**Reply by teoli2003** [reply]
Side note: Except maybe in the `alert()` page itself, we shouldn't use `alert()` at all. We have removed quite a lot of uses of it over the last year. If you find more, please fill an issue (or a PR) to get rid of `alert()`.

**Reply by wbamberg** [reply]
We have an issue for this work: https://github.com/mdn/content/issues/7566 . @AnilSeervi fixed a ton of them.

(although I'm not sure they are still broken in Chrome)

**Reply by OnkarRuikar** [reply]
Can we display console logs in live samples like StackOverflow does in it's runnable snippets? 
For example, https://stackoverflow.com/a/71456131/15273968

A little log window growing below the output box maybe.

**Comment by pepelsbey** [comment]
Closing as outdated/irrelevant. Please let me know if that’s not the case.
